api/types/authentication.go

```diff
@@ -722,7 +726,7 @@ func (c *AuthPreferenceV2) CheckAndSetDefaults() error {
 		c.Spec.SecondFactor = constants.SecondFactorWebauthn
 	case "":
 		// default to OTP if SecondFactors is also not set.
-		if len(c.Spec.SecondFactors) == 0 {
+		if c.Spec.SecondFactors == nil {
```

Comment:
The `len` check is usually better, `[]SecondFactorType{}` is not nil but it has zero length.

Reply:
Oops, I really thought this wasn't the case. I guess only `var sfs []SecondFactorType` would evaluate to nil.